#297 enchance map

RizzoMichael opened the pull request.

LucasBrazi06 commented:
No issue connected to this PR.
Should be in Dev cross review with Alix as reviewer.
Thanks!

AlixH commented on src/utils/Utils.tsx:
```diff
@@ -882,4 +890,49 @@ export default class Utils {
     return Utils.getLanguageFromLocale(Utils.getDeviceLocale());
   }
+
+  public static markerStatusChargingStations(connectorTab: Connector[], inactive: boolean) {
```
connectors instead of connectorTab

AlixH commented on src/utils/Utils.tsx:
```diff
@@ -882,4 +890,49 @@ export default class Utils {
+      connectorTab.find((connector) => connector.status === ChargePointStatus.FINISHING) ||
+      connectorTab.find((connector) => connector.status === ChargePointStatus.PREPARING)
+    ) {
+      return charging_finishing;
```
Maybe give a name for both preparing and finishing to the marker image

RizzoMichael replied:
Do you want me to change the name of the image?

AlixH commented on src/utils/Utils.tsx:
```diff
@@ -882,4 +890,49 @@ export default class Utils {
+      connectorTab.find((connector) => connector.status === ChargePointStatus.CHARGING) ||
+      connectorTab.find((connector) => connector.status === ChargePointStatus.OCCUPIED)
+    ) {
+      return charging_charging;
```
same as previous comment for charging/occupied

AlixH commented on src/utils/Utils.tsx:
```diff
@@ -19,8 +19,16 @@ import { RequestError } from '../types/RequestError';
+import charging_faulted from '../../assets/icon/charging_station_faulted.png';
+import charging_finishing from '../../assets/icon/charging_station_finishing.png';
+import charging_suspended from '../../assets/icon/charging_station_suspended.png';
+import charging_unavailable from '../../assets/icon/charging_station_unavailable.png';
```
Rename the files statusMarkerXXXXX with XXXX the status

RizzoMichael replied:
I didn't understand very well. Do you want me to rename the imports ?

AlixH commented on src/utils/Utils.tsx:
```diff
@@ -882,4 +890,49 @@ export default class Utils {
+    }
+  }
+
+  public static markerStatusSiteArea(siteArea: SiteArea) {
+    if (siteArea.connectorStats.availableConnectors > 0) {
+      return charging_available;
+    } else if (siteArea.connectorStats.chargingConnectors > 0) {
+      return charging_charging;
+    } else if (siteArea.connectorStats.unavailableConnectors > 0) {
+      return charging_unavailable;
+    }
+  }
+
+  public static markerStatusSite(site: Site) {
+    if (site.connectorStats.availableConnectors > 0) {
+      return charging_available;
+    } else if (site.connectorStats.chargingConnectors > 0) {
+      return charging_charging;
+    } else if (site.connectorStats.unavailableConnectors > 0) {
+      return charging_unavailable;
+    }
+  }
```
The methods are identical. I think you can pass a parameter in a single method to do both cases

jerome-benoit reviewed (changes requested):
You only need to use the vectorised images and remove the .png binary blobs from the tree. 

We should remove all .png images usage in fact for .svg.

AlixH commented:
When we open the map and then click on the arrow to go back, I would expect that it goes back to the list, not to the previous list. E.g, on Sites, if we open the map and then use the goes back arrow, we are redirected to Home.

AlixH commented on src/types/Server.tsx:
```diff
@@ -244,6 +243,10 @@ export enum ServerAction {
 
+  BOOT_NOTIFICATION = 'BootNotification',
+
+  METER_VALUES = 'MeterValues',
```
METERVALUES is used ?
It's the same as METER_VALUES

jerome-benoit replied:
@RizzoMichael have mis-merged with latest master-qa, I've cleaned up the mess in that branch with him.

AlixH commented:
> You only need to use the vectorised images and remove the .png binary blobs from the tree.
> 
> We should remove all .png images usage in fact for .svg.

@jerome-benoit , we tested together and it does not work with SVG

AlixH commented:
> You only need to use the vectorised images and remove the .png binary blobs from the tree.
> 
> We should remove all .png images usage in fact for .svg.

AlixH commented on src/screens/charging-stations/list/ChargingStations.tsx:
```diff
@@ -184,6 +185,8 @@ export default class ChargingStations extends BaseAutoRefreshScreen<Props, State
     if (this.siteAreaID) {
       // Go Back
       this.props.navigation.goBack();
+    } else if (this.state.showMap && !Utils.isEmptyArray(this.state.chargingStations)) {
+      this.setState({ showMap: false });
```
I think you can remove the line 187 and  just have if/else like on siteAreas but wrapped in the if(this.siteAreaID)

jerome-benoit commented:
@RizzoMichael: do not forget to update the PR status once you have addressed review comments and ask again for the review to the relevant people. 

Thanks.

RizzoMichael commented:
> @RizzoMichael: do not forget to update the PR status once you have addressed review comments and ask again for the review to the relevant people.
> 
> Thanks.

In this case I have to put or issue ?

jerome-benoit commented:
> > @RizzoMichael: do not forget to update the PR status once you have addressed review comments and ask again for the review to the relevant people.
> > Thanks.
> 
> Where do you want me put issue ?

To the right pipeline. PR have lifecycle to follow before their merge.

AlixH commented:
**Note:** Build is passing locally, since Stripe is merged build-ios fails

jerome-benoit commented on src/screens/charging-stations/list/ChargingStations.tsx:
```diff
@@ -345,6 +345,9 @@ export default class ChargingStations extends BaseAutoRefreshScreen<Props, State
       refreshing
     } = this.state;
     const mapIsDisplayed = showMap && !Utils.isEmptyArray(this.state.chargingStations);
+    const chargingStationsWithAddress = chargingStations.filter((chargingStation) =>
```
It's more `chargingStationsWithGPSCoordinates`

jerome-benoit commented on src/screens/site-areas/SiteAreas.tsx:
```diff
@@ -262,8 +267,9 @@ export default class SiteAreas extends BaseAutoRefreshScreen<Props, State> {
-    const { loading, skip, count, limit, initialFilters, showMap, siteAreaSelected } = this.state;
+    const { loading, skip, count, limit, initialFilters, showMap, siteAreaSelected, siteAreas } = this.state;
     const mapIsDisplayed = showMap && !Utils.isEmptyArray(this.state.siteAreas);
+    const siteAreaWithAddress = siteAreas.filter((siteArea) => Utils.containsAddressGPSCoordinates(siteArea.address));
```
`siteAreasWithGPSCoordinates`

jerome-benoit commented on src/screens/sites/Sites.tsx:
```diff
@@ -260,6 +265,7 @@ export default class Sites extends BaseAutoRefreshScreen<Props, State> {
     const { navigation } = this.props;
     const { loading, skip, count, limit, initialFilters, showMap, siteSelected, refreshing, sites } = this.state;
     const mapIsDisplayed = showMap && !Utils.isEmptyArray(this.state.sites);
+    const siteWithAddress = sites.filter((site) => Utils.containsAddressGPSCoordinates(site.address));
```
`sitesWithGPSCoordinates`

LucasBrazi06 commented on src/utils/Utils.tsx:
```diff
@@ -898,4 +905,41 @@ export default class Utils {
     return Utils.getLanguageFromLocale(Utils.getDeviceLocale());
   }
+
+  public static buildChargingStationStatusMarker(connectors: Connector[], inactive: boolean) {
```
Missing return type

LucasBrazi06 commented on src/utils/Utils.tsx:
```diff
@@ -898,4 +905,41 @@ export default class Utils {
+    }
+  }
+
+  public static buildSiteStatusMarker(connectorStats: ConnectorStats) {
```
Missing return type

LucasBrazi06 commented on src/utils/Utils.tsx:
```diff
@@ -23,6 +23,13 @@ import { InactivityStatus } from '../types/Transaction';
 import User, { UserRole, UserStatus } from '../types/User';
 import Constants from './Constants';
 import Message from './Message';
+import statusMarkerAvailable from '../../assets/icon/charging_station_available.png';
```
Use postfix Png.
statusMarkerAvailable -> statusMarkerAvailablePng
So we know that this is an image when we read the code.

AlixH commented:
The map is still blank (might be an API key issue)
